server/src/main/java/org/apache/iotdb/db/query/factory/AggregateResultFactory.java

```diff
@@ -134,4 +135,36 @@ public static AggregateResult getAggrResultByType(
         throw new IllegalArgumentException("Invalid Aggregation Type: " + aggregationType.name());
     }
   }
+
+  public static SlidingWindowAggrQueue getSlidingWindowAggrQueueByName(
+      String aggrFuncName, TSDataType dataType, boolean ascending) {
+    if (aggrFuncName == null) {
+      throw new IllegalArgumentException("AggregateFunction Name must not be null");
+    }
+
+    switch (aggrFuncName.toLowerCase()) {
+      case SQLConstant.SUM:
+      case SQLConstant.AVG:
+      case SQLConstant.COUNT:
+        return new SumAvgCountSlidingWindowAggrQueue(dataType, aggrFuncName, ascending);
```

Comment:
These queue names are very strange. If I add another aggreateResult named `X` in the furture, which words the name will be changed to? 
I think it is better to add an  **abstract** method called 'constructAggrQueue' in `AggregateResult`. Each sub classes will implement it and return a new `AggrQueue`. The aggr queue can code as an innter class of each aggregation result.  Then this method is not needed any more. This is a OOP style design.

Reply:
It also think it's better to do so. And in the meanwhile, some Aggregation function share the same logic, if we implement it as an inner class of each aggregation result, we also need to figure out a way to avoid code duplication. Any good ideas about how to avoid these code duplication? @ericpai

Reply:
I haven't investigated too deeply of the differences between these queues yet. Maybe they can be placed in a separate package and renamed with an aggregate-name-free name first, and each `constructAggrQueue` implementation just codes as ` return new XXXQueue()`. After that, let's see how to refine these queue implementations themselves.

Reply:
I renamed these classes with the algorithm description instead of the aggregate function, maybe this problem doesn't exist now.